assign-doi-datacite/src/main/java/no/unit/nva/doi/datacite/clients/MdsClient.java

```diff
@@ -0,0 +1,178 @@
+    private final URI dataciteMdsUri;
+    private final CustomerConfigExtractor customerConfigExtractor;
+
+    public MdsClient(URI dataciteMdsUri,
+                     CustomerConfigExtractor customerConfigExtractor,
+                     HttpClient httpClient) {
+        super(httpClient);
+        this.dataciteMdsUri = dataciteMdsUri;
+        this.customerConfigExtractor = customerConfigExtractor;
+    }
+
+    public void updateMetadata(URI customerId, Doi doi, String metadataDataCiteXml)
+        throws ClientException {
+        var customer = customerConfigExtractor.getCustomerConfig(customerId);
+        validateUpdateMetadataInput(doi, metadataDataCiteXml);
+        var request = createPutMetadataRequest(customer, doi, metadataDataCiteXml);
+        sendRequest(request,  HttpStatus.SC_OK);
+    }
+
+    public void setLandingPage(URI customerId, Doi doi, URI landingPage) throws ClientException {
+        var customer = customerConfigExtractor.getCustomerConfig(customerId);
+        validateLandingPageInput(doi, landingPage);
+        var request = createLandingPagePostRequest(customer, doi, landingPage);
+        sendRequest(request,  HttpStatus.SC_CREATED);
+    }
+
+    public void deleteMedata(URI customerId, Doi doi) throws ClientException {
+        var customer = customerConfigExtractor.getCustomerConfig(customerId);
+        validateDeleteMetadataRequest(doi);
+        var request = createDeleteMetadataRequest(customer, doi);
+        sendRequest(request, HttpStatus.SC_OK);
+    }
+
+    public void deleteDraftDoi(URI customerId, Doi doi) throws ClientException {
+        var customer = customerConfigExtractor.getCustomerConfig(customerId);
+        validateDeleteDraftDoiRequest(doi);
+        var request = createDeleteDraftDoiRequest(customer, doi);
+        sendDeleteDraftRequest(request, doi);
+    }
+
+    private void sendDeleteDraftRequest(HttpRequest request, Doi doi) throws ClientException {
+        var response = attempt(() -> super.getHttpClient().send(request, BodyHandlers.ofString()))
+                           .orElseThrow(this::handleFailure);
+        if (response.statusCode() == HttpStatus.SC_METHOD_NOT_ALLOWED) {
+            logger.error(REQUEST_RESPONDED_WITH_RESPONSE_MESSAGE + response.body());
+            throw new DeleteDraftDoiException(doi, response.statusCode());
+        }
+        if (response.statusCode() != HttpStatus.SC_OK) {
+            logger.error(REQUEST_RESPONDED_WITH_RESPONSE_MESSAGE + response.body());
+            throw new ClientException(response.toString());
+        }
+    }
+
+    private HttpRequest createDeleteMetadataRequest(CustomerConfig customer, Doi doi)
+        throws CustomerConfigException {
+        return HttpRequest.newBuilder()
+                   .DELETE()
+                   .uri(createUriForUpdatingMetadata(doi))
+                   .header(AUTHORIZATION_HEADER, customer.extractBasicAuthenticationString())
```

Comment:
i see you are setting auth header manually on every request. is there a reason you are not using an authenticated http client? is it because the client is shared between open and authorized requests?